subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -81,10 +81,14 @@ inline fun <reified T : Plugin<Project>> Project.apply() =
  * Executes the given configuration block against the [plugin convention]
  * [Convention.getPlugin] or extension of the specified type.
  *
+ * Note, that the concept of conventions is deprecated and scheduled for
+ * removal in Gradle 8.
+ *
  * @param T the plugin convention type.
  * @param configuration the configuration block.
  * @see [Convention.getPlugin]
  */
+@Suppress("deprecation")
 inline fun <reified T : Any> Project.configure(noinline configuration: T.() -> Unit): Unit =
     typeOf<T>().let { type ->
```

Comment:
💄 Move the suppression closer to the deprecated member usage
```suggestion
inline fun <reified T : Any> Project.configure(noinline configuration: T.() -> Unit): Unit =
    @Suppress("deprecation")
    typeOf<T>().let { type ->
```